libs/components/src/lib/menu-item/menu-item.template.ts

```diff
@@ -12,43 +14,62 @@ const getCheckIcon = (affixIconTemplate: any, x: MenuItem, iconType: string) =>
 };
 
 const getClasses = ({
-	disabled, checked, expanded, role, text, textSecondary, icon, metaSlottedContent
-}: MenuItem) =>	classNames(
+	disabled, checked, role, text, textSecondary, icon, metaSlottedContent
+}: MenuItem) => classNames(
 	'base',
 	['disabled', Boolean(disabled)],
 	['selected', role !== MenuItemRole.menuitem && Boolean(checked)],
 	['trailing', role !== MenuItemRole.menuitem && Boolean(icon)],
-	['expanded', Boolean(expanded)],
 	['item-checkbox', role === MenuItemRole.menuitemcheckbox],
 	['item-radio', role === MenuItemRole.menuitemradio],
 	['two-lines', Boolean(text?.length) && Boolean(textSecondary?.length)],
 	['has-meta', Boolean(metaSlottedContent?.length)]
 );
 
-function handleClick(x: MenuItem, { event } : ExecutionContext<MenuItem>) {
+function handleClick(x: MenuItem, { event }: ExecutionContext<MenuItem>) {
 	x.handleMenuItemClick(event as MouseEvent);
 	return (x as any).role === MenuItemRole.presentation;
 }
+
+function checkbox(context: ElementDefinitionContext) {
+	const affixIconTemplate = affixIconTemplateFactory(context);
+
+	return html<MenuItem>`${when(x => x.role === MenuItemRole.menuitemcheckbox,
+		html`<span class="action">${x => getCheckIcon(affixIconTemplate, x, 'checkbox')}</span>`)}`;
+}
+
+function radio(context: ElementDefinitionContext) {
+	const affixIconTemplate = affixIconTemplateFactory(context);
+
+	return html<MenuItem>`${when(x => x.role === MenuItemRole.menuitemradio,
+		html`<span class="action">${x => getCheckIcon(affixIconTemplate, x, 'radio')}</span>`)}`;
+}
+
+function text() {
+	return html<MenuItem>`${when(x => x.text || x.textSecondary,
+		html`<span class="text">
+			${when(x => x.text, html`<span class="text-primary">${x => x.text}</span>`)}
+			${when(x => x.textSecondary, html`<span class="text-secondary">${x => x.textSecondary}</span>`)}
+		</span>`)}`;
+}
+
 /**
  * Generates a template for the (MenuItem:class) component using
  * the provided prefix.
  *
  * @param context - element definition context
  * @public
  */
-export const MenuItemTemplate:  (
-	context: ElementDefinitionContext,
-	definition: MenuItemOptions
-) => ViewTemplate<MenuItem> = (
-	context: ElementDefinitionContext,
-	definition: MenuItemOptions
-) => {
+export const MenuItemTemplate: (context: ElementDefinitionContext, definition: MenuItemOptions
+) => ViewTemplate<MenuItem> = (context: ElementDefinitionContext) => {
 	const affixIconTemplate = affixIconTemplateFactory(context);
 	const focusTemplate = focusTemplateFactory(context);
+	const iconTag = context.tagFor(Icon);
 
 	return html<MenuItem>`
 	<template
 		role="${x => (x.role ? x.role : MenuItemRole.menuitem)}"
+		aria-haspopup="${x => (x.hasSubmenu ? 'menu' : void 0)}"
 		aria-checked="${x => (x.role !== MenuItemRole.menuitem ? x.checked : void 0)}"
```

Comment:
Not sure who wrote `void 0` to begin with, but let's use undefined instead

```suggestion
		aria-haspopup="${x => (x.hasSubmenu ? 'menu' : undefined)}"
		aria-checked="${x => (x.role !== MenuItemRole.menuitem ? x.checked : undefined)}"
```

Reply:
@RichardHelm The void 0 is copied from FAST code